sequencer/closingchecker.go

```diff
@@ -66,23 +66,15 @@ func (s *Sequencer) shouldCloseDueToNewDeposits(ctx context.Context) (bool, erro
 	return false, nil
 }
 
-// shouldCloseTooLongSinceLastVirtualized returns true if last batch virtualization happened
+// shouldCloseTooLongSinceLastSequence returns true if last batch virtualization happened
 // more than MaxTimeForBatchToBeOpen ago and there are transactions in the current sequence
-func (s *Sequencer) shouldCloseTooLongSinceLastVirtualized(ctx context.Context) (bool, error) {
-	lastBatchNumber, err := s.state.GetLastBatchNumber(ctx, nil)
-	if err != nil {
-		log.Errorf("failed to get last batch number, err: %w", err)
-		return false, err
-	}
-	isPreviousBatchVirtualized, err := s.state.IsBatchVirtualized(ctx, lastBatchNumber-1, nil)
-	if err != nil {
-		log.Errorf("failed to get last virtual batch num, err: %w", err)
-		return false, err
-	}
-	if time.Unix(s.sequenceInProgress.Timestamp, 0).Add(s.cfg.MaxTimeForBatchToBeOpen.Duration).Before(time.Now()) &&
-		isPreviousBatchVirtualized && len(s.sequenceInProgress.Txs) > 0 {
+func (s *Sequencer) shouldCloseTooLongSinceLastSequence(ctx context.Context) bool {
+	timeoutExpired := s.sequenceInProgress.Timestamp.Add(s.cfg.MaxTimeForBatchToBeOpen.Duration).Before(time.Now())
```

Comment:
This check is not working as expected. The batch could be open any time long, but the time from the previous virtualized batch shouldn't be more, than a time.Now()-value from the config.
https://github.com/0xPolygonHermez/zkevm-node/issues/1245

Reply:
the sequencer logic was changed to not depend on virtualized batches, now it just depends on the sequences created in the DB.